docs/usage/interpreter/interpreter_binding_mode.md

```diff
@@ -25,4 +25,97 @@ limitations under the License.
 
+<br/>
+
+Interpreter is a JVM process that communicates with Zeppelin daemon using thrift. 
+Each Interpreter process can have a interpreter group, and each interpreter instance belongs to this interpreter group.
+(See [here](../../development/writing_zeppelin_interpreter.html) to understand more about its internal structure.) 
+
+Zeppelin provides 3 different modes to run interpreter process: **shared**, **scoped** and **isolated**.   
+Also, user can specify the scope of these mode as well: **per user** or **per note**.  
+These 3 modes give flexibility to fit Zeppelin into any type of use cases.
+
+In this documentation, we mainly discuss the combination of **per note** mode with **shared**, **scoped** and **isolated** modes for explanation. 
+
+## Shared Mode
+
+<div class="text-center">
+    <img src="{{BASE_PATH}}/assets/themes/zeppelin/img/docs-img/interpreter_binding_mode-shared.png" height="40%" width="40%">
+</div>
+<br/>
+
+In **Shared** mode, single JVM process and a single session serves all notes. As a result, `note A` can access variables (e.g python, scala, ..) directly created from other notes.. 
+
+## Scoped Mode
+
+<div class="text-center">
+    <img src="{{BASE_PATH}}/assets/themes/zeppelin/img/docs-img/interpreter_binding_mode-scoped.png">
+</div>
+<br/>
+
+In Scoped mode, Zeppelin still runs single interpreter JVM process but multiple sessions serve each note. (in case of **per note**) 
+So, each note have their own dedicated session. (but still possible to share objects via [ResourcePool](../../interpreter/spark.html#object-exchange)) 
+
+## Isolated Mode
+
+<div class="text-center">
+    <img src="{{BASE_PATH}}/assets/themes/zeppelin/img/docs-img/interpreter_binding_mode-isolated.png">
+</div>
+<br/>
+
+**Isolated** mode runs separate interpreter process for each note. (in case of **per note**) 
+So, each note have absolutely isolated session. (but still possible to share objects via [ResourcePool](../../interpreter/spark.html#object-exchange)) 
+
+## Which mode should I use?
+
+<br/>
+
+Mode | Each notebook...	| Benefits | Disadvantages | Sharing objects
+--- | --- | --- | --- | ---
+**shared** | Shares a single sessions in a single interpreter process (JVM) | Low resource utilization and Easy to share data between notebooks | All notebooks are affected if Interpreter Process dies | can share directly
+**scoped** | Has its own note sessions in the same Interpreter Process (JVM) | Less resource utilization than isolated mode | All notebooks are affected if Interpreter Process dies | can't share directly, but possible to share objets via [ResourcePool](../../interpreter/spark.html#object-exchange)) 
+**isolated** | Has its own Interpreter Process | One notebook not affected directly by other notebooks (**per note**) | Can't share data between notebooks easily (**per note**) | can't share directly, but possible to share objets via [ResourcePool](../../interpreter/spark.html#object-exchange)) 
```

Comment:
Can't share directly, but it's possible to share objects via [ResourcePool](../../interpreter/spark.html#object-exchange)